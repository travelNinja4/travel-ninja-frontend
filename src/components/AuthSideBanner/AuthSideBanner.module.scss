/**
------------------------------------------------------
styles specific to the AuthSideBanner component
------------------------------------------------------
**/

.container {
  position: relative;
  width: 100%;
  height: 100%;
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: center;
}

.imgContainer {
  object-fit: cover;
  z-index: 0;
}

.overlay {
  position: absolute;
  inset: 0;
  background: linear-gradient(135deg, var(--color-indigo) 0%, var(--color-purple) 100%);
  opacity: 0.6;
  z-index: 1;
}

.headerContainer {
  position: relative;
  z-index: 2;
  text-align: center;
  color: var(--color-white);

  &.header {
    font-size: 3rem;
    font-weight: 700;
    margin-bottom: 1rem;
    text-shadow: 0 4px 8px var(--color-black-alpha-30);
  }

  &.subHeader {
    font-size: 1.2rem;
    opacity: 0.9;
    font-weight: 300;
  }
}
