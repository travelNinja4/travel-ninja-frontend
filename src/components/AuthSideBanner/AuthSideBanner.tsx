/**
 * AuthSideBanner is a reusable UI component that displays a full-height side banner with an optional background image, overlay, title, and description.
 *
 * @example
 * ```tsx
 * import AuthSideBanner from '@src/components/AuthSideBanner'
 *
 * export default function AuthSideBanner() {
 *   return <AuthSideBanner label="Hello" />;
 * }
 * ```
 */

import { STRINGS } from '@/constants/strings';
import CustomImage from '../CustomImage';
import Typography from '../Typography';
import styles from './AuthSideBanner.module.scss';

interface AuthSideBannerProps {
  appTagLine: string;
}

export default function AuthSideBanner({ appTagLine }: AuthSideBannerProps) {
  return (
    <div data-testid="AuthSideBannerTest" className={styles.container}>
      <CustomImage
        isUrl
        src="https://picsum.photos/1920/1080?random=1"
        priority
        alt="authImg"
        className={styles.imgContainer}
        unoptimized
        fill
      />
      <div className={styles.overlay}></div>
      <div className={styles.headerContainer}>
        <Typography tag="h1" className={styles.header} color="#ffffff" align="center">
          {STRINGS.APP_NAME}
        </Typography>
        <Typography tag="p" className={styles.subHeader} color="#ffffff" align="center">
          {appTagLine}
        </Typography>
      </div>
    </div>
  );
}
