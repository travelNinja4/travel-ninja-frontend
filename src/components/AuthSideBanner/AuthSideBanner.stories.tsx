import type { Meta, StoryObj } from '@storybook/nextjs-vite';
/**
import { fn } from 'storybook/test';
*/
import AuthSideBanner from './AuthSideBanner'

const meta = {
  title: 'components/AuthSideBanner',
  component: AuthSideBanner,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    label: { 
        control: 'text', 
        description: 'Sample label for the component' 
        },
  },
  args: { 
    /** onClick: fn() **/
    },
} satisfies Meta<typeof AuthSideBanner>;

export default meta;
type Story = StoryObj<typeof meta>;