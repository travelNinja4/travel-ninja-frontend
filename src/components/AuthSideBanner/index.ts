import AuthSideBanner from './AuthSideBanner';

export default AuthSideBanner;
