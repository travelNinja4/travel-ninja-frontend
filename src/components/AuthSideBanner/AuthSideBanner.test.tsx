import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import AuthSideBanner from './AuthSideBanner';

describe('Home', () => {
  /** Base props for AuthSideBanner **/
  const baseProps = {
    label: 'AuthSideBanner Test',
  };

  /**
   * @test
   * Renders the AuthSideBanner component without throwing an error
   * and checks if the AuthSideBannerTest text is present in the document.
   *
   * This ensures that the component loads correctly with the base props.
   */
  it('renders without crashing', () => {
    render(<AuthSideBanner {...baseProps} />);
    expect(screen.getByTestId('AuthSideBannerTest')).toBeInTheDocument();
  });

  /**
   * @test
   * Ensures that the AuthSideBanner component renders without throwing any runtime errors.
   *
   * This is a safety check to confirm that the component can mount successfully
   * with the provided base props, even if it does not render any visible content.
   */
  it('renders without throwing', () => {
    expect(() => render(<AuthSideBanner {...baseProps} />)).not.toThrow();
  });

  /**
   * @test
   * Captures a snapshot of the AuthSideBanner component with base props
   * and ensures it matches the previously stored snapshot.
   *
   * This helps detect any unexpected UI changes.
   */
  it('matches snapshot', () => {
    const { asFragment } = render(<AuthSideBanner {...baseProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
