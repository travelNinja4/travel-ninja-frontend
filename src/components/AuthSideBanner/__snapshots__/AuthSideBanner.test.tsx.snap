// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Home matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-testid="AuthSideBannerTest"
  >
    <img
      alt="authImg"
      blurdataurl="https://res.cloudinary.com/dyci0wg0i/image/upload/fallbackImg.png"
      class="imgContainer"
      data-testid="CustomImageTest"
      placeholder="blur"
      src="https://picsum.photos/1920/1080?random=1"
    />
    <div
      class="overlay"
    />
    <div
      class="headerContainer"
    >
      <h1
        class="typography h1 align-center header"
        data-testid="TypographyTest"
        style="color: var(--color-white);"
      >
        TravelNinja
      </h1>
      <p
        class="typography p align-center subHeader"
        data-testid="TypographyTest"
        style="color: var(--color-white);"
      >
        AuthSideBanner Test
      </p>
    </div>
  </div>
</DocumentFragment>
`;
